### DISCUSSION TITLE
Help setting up designer in existing Angular App

### DISCUSSION THREAD

**Discussion by rjps12** [elsa-workflows/elsa-core#2815] [Q&A]
Hi, I'm trying to follow the example of setting up the designer from here [https://codesandbox.io/s/angular-elsa-designer-dkb4t?file=/src/assets/workflow.json](https://codesandbox.io/s/angular-elsa-designer-dkb4t?file=/src/assets/workflow.json)

I'm trying to build something like the user can create their own workflow then save it to the database.

Questions:

1. How to initialize the designer with an empty workflow? Seems like it's required to have a json file when initializing the designer.
2. How to save it to the database? 

I cannot save anything nor can I find anything in guides on your webpage about making it work.

How do I communicate with the backend part that has a database connection and should run these flows?

When I make these flows, how am I running them?

**Comment by sfmskywalker** [comment]
Hi @rjps12 

The Elsa 2 designer only works when configured to point to an Elsa server, which means you can't used it on its own.
Once configured to point to an Elsa server, you don't have to do anything for loading & saving workflows.

The Elsa 3 designer (currently under development) on the other hand does allow you to run in "disconnected" mode, giving you full programmatic control over what workflows to initialize it with (optional) and reading the current workflow state so that you can control yourself what to do with it.
